### DISCUSSION TITLE
Props are not resolving correctly with [styled-components]

### DISCUSSION THREAD

**Discussion by lifeiscontent** [DefinitelyTyped/DefinitelyTyped#58250] [Issues with a @types package]
Here's my reproduction:

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAZxgTwDYFMAmAaOBvOAYwQVwGUUNMBhCcCAO3QZgAUoIwE4BfOAMw4g4AIiRosAWkJ1ITFghEBuAFArgLdFH4BDQujgAhAK4wYjdp254VcRMABe6AFxwA1umQR+cFGHTe9k4IqjxqMgxIQejcALz4toggOqiorsQIAAYAFujAAObZMK4ATAAMYAAeSpnYiSBYwMYg6SQ5eYXFcADMFdW1iag6UPkuRG25BUWuACx9NXV8OtwRSKoqq-AmZoxw8eJUAHQARqbmDAA82+eWXAB8mYl2ACR4ABRgHFwAlHt3cB8vghDghHAYAPzRBAAbU+VhBYIAunBXMYGJh0PwNFhvjxVJl1gB6QlwADu0DcCBUfgMAHUKcMIGjMLc4nAAEroPQwQ60ejyNhAi6JTncw4AMTRhBgwEYfLkzBgwrsdgoEhoskYitZFxEpx2DBEuB0DGQuGuFiBuCYADctHdEg67kSSccOB4GNTkP4jO7mKy9hyudLeZqBTqaYELQxnWoab6IB6yGDA4Y-QxWdCxGCRIilHBiXA3YnmF6ffSoG5Gczk05AxWqxxmZns05c-nC+TK1SgA

[a comment by an automated account is not shown]

**Reply by lifeiscontent** [reply]
@eps1lon any ideas here?

**Reply by eps1lon** [reply]
I think the issue is that you're using a host component (that has hardcoded types) but try to assume it has more types. We wouldn't actually consider `<Button sizes="small" />` sound since it would end up as `<button sizes="small" />`. 

I don't know how and if you could support that pattern. But I don't think we should allow it as is since we would also reject `<button sizes="small" />`.

**Reply by lifeiscontent** [reply]
@eps1lon that doesn't make sense. In the docs they use invalid host props as well, and automatically filter out any non-host attributes that seemed like the whole point of styled components so why wouldn't it be supported?

**Reply by eps1lon** [reply]
The way we intended the behavior of the first type parameter doesn't allow what you're looking for. The props should match the component type but you want that they can have no intersection at all. 

> that doesn't make sense

Did I not express myself clearly or do you just disagree with the current state? Note that allowing `<Button sizes="small" />` would not match type-checking for host components and could be confusing.

**Reply by lifeiscontent** [reply]
@eps1lon on the homepage of https://styled-components.com/ they are using a prop called `primary` which is not an html attribute (host component prop) are we mixing terms here? because if the homepage of styled components says `primary` can be a prop, so should the types.
